rust/kcl-lib/src/execution/geometry.rs

```diff
@@ -1819,16 +1819,19 @@ pub enum SegmentKind {
     Point {
         position: [TyF64; 2],
         ctor: Box<PointCtor>,
-        freedom: Freedom,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        freedom: Option<Freedom>,
     },
     Line {
         start: [TyF64; 2],
         end: [TyF64; 2],
         ctor: Box<LineCtor>,
         start_object_id: ObjectId,
         end_object_id: ObjectId,
-        start_freedom: Freedom,
-        end_freedom: Freedom,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        start_freedom: Option<Freedom>,
```

Comment:
Are these two Option<Freedom> either both Some or both None? If so, maybe break this into a LineFreedom struct with two mandatory fields, and use Option<LineFreedom>. Same for Arc below